### PR TITLE
Adding counter contract with access control

### LINKED ISSUE #28: Create a Counter Contract but with access control
## Description
- We want to have an example of access control to some methods, this can be done with a simple CounterContract.

## Steps
- Create a CounterContract 
- Add access control to the methods so that the owner of the contract can call them.
- Add unit tests.

### PR DESCRIPTION
**Pull request by Gerson2102**
## Issue that Solves:
Closes #28 

## Main changes:

This PR is adding a simple counter contract with some access control. Only owner can call certain functions. And I added some additional functionalities related to addresses because are part of the workflow with addresses.

## Screenshot of behaviour:

Everything is working on my side:

image

### CODE REVIEW

**Comment by danielcdz** [comment]
@Gerson2102 this looks good overall but please move it to the contract directory inside `/starknet`

**Comment by Gerson2102** [comment]
Got it. I think I addressed all the feedback now.

File: cairo/scripts/.tool-versions
Comment (danielcdz) [inline cairo/scripts/.tool-versions:1]: this is not needed

File: cairo/scripts/hello_world/.tool-versions
Comment (danielcdz) [inline cairo/scripts/hello_world/.tool-versions:1]: can you remove all changes from this directory please

**Review by danielcdz — CHANGES_REQUESTED** [review]

File: cairo/scripts/hello_world/.tool-versions
Comment (Gerson2102) [reply cairo/scripts/hello_world/.tool-versions:1]: Which changes? Deleting `.tool-versions` and `Scarb.lock`?

Comment (Gerson2102) [reply cairo/scripts/hello_world/.tool-versions:1]: Or deleting just the contents of `.tool-versions`?

Comment (danielcdz) [reply cairo/scripts/hello_world/.tool-versions:1]: you have changes in cairo/scripts/hello_world that are not needed, don't include those changes in this PR

Comment (Gerson2102) [reply cairo/scripts/hello_world/.tool-versions:1]: I deleted the versions file and the changes in the scarb lock.

**Review by danielcdz — APPROVED** [review]

**Pull request merged**